src/APIs/RedisAPI.ts

```diff
@@ -11,49 +16,75 @@ export const getCachedWords = async ({ key, redisClient }) => {
   return cachedWords;
 };
 
-export const setCachedWords = async ({
-  key,
-  data,
-  redisClient,
-  version,
-}) => {
-  const updatedData = assign(data);
+interface SetCachedWordsParams {
+  key: string;
+  data: any; // Update the type of `data` accordingly
+  redisClient: RedisClient;
+  version: string; // Update the type of `version` accordingly
+}
+
+export const setCachedWords = async ({ key, data, redisClient, version }: SetCachedWordsParams) => {
+  const updatedData = assign({}, data);
   updatedData.words = minimizeWords(data.words, version);
   if (!redisClient.isFake) {
     await redisClient.set(key, JSON.stringify(updatedData), { EX: REDIS_CACHE_EXPIRATION });
   }
   return updatedData;
 };
```

Comment:
I think explicitly stating the return types would give the full benefit of typescript.  The function returns and `updatedData` but there's no return type to it. The benefit of this is that where ever it called we would know what to expect

Reply:
hmm I'm not entirely convinced tbh, the type is inferred from `data` which should be defined within `SetCachedWordsParams`. explicitly defining the return type of the function is more for readability purposes, which you can argue is important in this case - but i can go either way

@shaikahmadnawaz can you do `assign(data)` instead?